### PR TITLE
dev-java/lucene: switch dependency to dev-java/ant

### PR DESCRIPTION
**Pull request by vaukai**
These are some of the packages which still depend on `ant-core`.

### CODE REVIEW

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

File: dev-java/iso-relax/iso-relax-20050331-r6.ebuild [hunk 1]
```diff
@@ -10,30 +10,23 @@ inherit java-pkg-2 java-pkg-simple
...
 DEPEND="
-	${CDEPEND}
-	>=virtual/jdk-1.8:*"
-
-RDEPEND="
-	${CDEPEND}
-	>=virtual/jre-1.8:*"
-
-S="${WORKDIR}/${P}"
+	dev-java/ant:0
```
Comment (fordfrog) [inline dev-java/iso-relax/iso-relax-20050331-r6.ebuild:20]: this package also seems to need `ant` during runtime:
```
$ grep -i \\.ant * -r
src/org/iso_relax/ant/ErrorHandlerImpl.java:package org.iso_relax.ant;
src/org/iso_relax/ant/ErrorHandlerImpl.java:import org.apache.tools.ant.Project;
src/org/iso_relax/ant/ErrorHandlerImpl.java:import org.apache.tools.ant.Task;
src/org/iso_relax/ant/JARVTask.java:package org.iso_relax.ant;
src/org/iso_relax/ant/JARVTask.java:import org.apache.tools.ant.BuildException;
src/org/iso_relax/ant/JARVTask.java:import org.apache.tools.ant.DirectoryScanner;
src/org/iso_relax/ant/JARVTask.java:import org.apache.tools.ant.Project;
src/org/iso_relax/ant/JARVTask.java:import org.apache.tools.ant.Task;
src/org/iso_relax/ant/JARVTask.java:import org.apache.tools.ant.types.FileSet;
```

**Review by fordfrog — APPROVED** [review]

File: dev-java/cglib/cglib-3.3.0-r2.ebuild [hunk 2]
```diff
@@ -16,26 +16,24 @@ MY_P=cglib-${MY_PV}
...
 LICENSE="Apache-2.0"
 SLOT="3"
 KEYWORDS="amd64 ~arm arm64 ppc64 x86"
 
-CDEPEND="dev-java/ant-core:0
-	dev-java/asm:9
-"
+CP_DEPEND="dev-java/asm:9"
 DEPEND="
+	dev-java/ant:0
```
[the remark this answers is not in the document]
Comment (vaukai) [reply dev-java/cglib/cglib-3.3.0-r2.ebuild]: done.

File: dev-java/iso-relax/iso-relax-20050331-r6.ebuild (hunk 1 above)
Comment (vaukai) [reply dev-java/iso-relax/iso-relax-20050331-r6.ebuild]: done.

**Comment by vaukai** [comment]
guess i've got them all now, at least from `::gentoo`
- `dev-lang/scala` got [another PR](https://github.com/gentoo/gentoo/pull/28435), which seems to not need ant any longer
-  [`gnuprologjava` and `interprolog`](https://github.com/gentoo/gentoo/pull/35724) in another PR
- `android-sdk-update-manager` and `mozart` in a [PR for proxy-maint](https://github.com/gentoo/gentoo/pull/35725)
- `plantuml` gets solved after stabilization
- [`libbluray`](https://github.com/gentoo/gentoo/pull/35711) and [`gdal`](https://github.com/gentoo/gentoo/pull/35165) in their own PRs
- `tomcat` gets frequent updates

watching https://packages.gentoo.org/packages/dev-java/ant-core/reverse-dependencies

**Comment by fordfrog** [comment]
i just updated tomcat ebuilds as i forgot to do that on the last bump

**Comment by vaukai** [comment]
> i just updated tomcat ebuilds as i forgot to do that on the last bump

for tomcat to include the `non-standalone` `bnd` i also have some early wip based on https://github.com/apache/tomcat/blob/9.0.87/build.xml#L227-L232

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

File: dev-java/ant-contrib/ant-contrib-1.0_beta6_pre20201123-r2.ebuild [hunk 3]
```diff
@@ -49,20 +47,18 @@ CDEPEND="
 DEPEND="${CDEPEND}
 	>=virtual/jdk-1.8:*
 	test? (
-		dev-java/ant-core:0
+		>=dev-java/ant-1.10.14-r3:0
```
[the remark this answers is not in the document]
Comment (vaukai) [reply dev-java/ant-contrib/ant-contrib-1.0_beta6_pre20201123-r2.ebuild:50]: > i guess `[junit4]` is missing here. might be needed in other ebuilds too, i checked just this one.

tests are restricted anyway. but I will check

**Comment by vaukai** [comment]
> @vaukai i merged some commits. i also added a review to `dev-java/ant-contrib`, i hope you can see it.

yep. this time it's visible for me.

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — COMMENTED** [review]
i can't test this one:
```
>>> Test phase: dev-java/lucene-2.4.1-r4
Adding gentoo.classpath to common-build.xml
Rewriting common-build.xml
 * Using following ANT_TASKS: ant-junit
!!! ERROR: Package ant-junit was not found!
 * ERROR: dev-java/lucene-2.4.1-r4::gentoo failed (test phase):
 *   java-config --classpath=ant-junit failed
 * 
 * Call stack:
 *     ebuild.sh, line  136:  Called src_test
 *   environment, line 3062:  Called eant 'test-core'
 *   environment, line  570:  Called java-pkg_getjars '--with-dependencies' 'ant-junit'
 *   environment, line 2165:  Called die
 * The specific snippet of code:
 *       [[ $? != 0 ]] && die "java-config --classpath=${pkgs} failed";
 * 
 * If you need support, post the output of `emerge --info '=dev-java/lucene-2.4.1-r4::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=dev-java/lucene-2.4.1-r4::gentoo'`.
!!! When you file a bug report, please include the following information:
GENTOO_VM=openjdk-bin-8  CLASSPATH="" JAVA_HOME="/opt/openjdk-bin-8.402_p06"
JAVACFLAGS="-source 1.8 -target 1.8" COMPILER="javac"
and of course, the output of emerge --info =lucene-2.4.1
 * The complete build log is located at '/var/tmp/portage/dev-java/lucene-2.4.1-r4/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/dev-java/lucene-2.4.1-r4/temp/environment'.
 * Working directory: '/var/tmp/portage/dev-java/lucene-2.4.1-r4/work/lucene-2.4.1'
 * S: '/var/tmp/portage/dev-java/lucene-2.4.1-r4/work/lucene-2.4.1'
```

**Comment by fordfrog** [comment]
btw, what's approved has been merged

**Comment by vaukai** [comment]
> i can't test this one:
> 
> ```
> >>> Test phase: dev-java/lucene-2.4.1-r4
> Adding gentoo.classpath to common-build.xml
> ```

i have to postpone this. need to get a better understanding of what eclasses ro with `ANT_TASKS=`.

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

**Comment by fordfrog** [comment]
again, what's approved is merged

**Review by fordfrog — APPROVED** [review]

**Comment by vaukai** [comment]
> i can't test this one:
> 
> ```
> >>> Test phase: dev-java/lucene-2.4.1-r4
> Adding gentoo.classpath to common-build.xml
> Rewriting common-build.xml
>  * Using following ANT_TASKS: ant-junit
> !!! ERROR: Package ant-junit was not found!
>  * ERROR: dev-java/lucene-2.4.1-r4::gentoo failed (test phase):
>  *   java-config --classpath=ant-junit failed
> ```

The same error happens also with the presently stable version `dev-java/lucene-2.4.1-r3`. 
It should have been handled within https://github.com/gentoo/gentoo/pull/34745/commits where we adjusted what we thought were **all** dependencies to `ant-junit` or `ant-junit4`. 
At that time we forgot to run
``` 
qgrep ANT_TASKS=
```
which should have shown `lucene` amongst some others. Affected so far is (presently) only lucene. Looking at how the `ANT_TASKS` variable is handled in the `eant()` function in `java-utils2.eclass` shows how it works: 

```
	export ANT_TASKS

	[[ -n ${JAVA_PKG_DEBUG} ]] && antflags="${antflags} --execdebug -debug"
	[[ -n ${PORTAGE_QUIET} ]] && antflags="${antflags} -q"

	local gcp="${EANT_GENTOO_CLASSPATH}"
	local getjarsarg=""

	if [[ ${EBUILD_PHASE} = "test" ]]; then
		antflags="${antflags} -DJunit.present=true"
		getjarsarg="--with-dependencies"

		local re="\bant-junit4?([-:]\S+)?\b"
		[[ ${ANT_TASKS} =~ ${re} ]] && gcp+=" ${BASH_REMATCH[0]}"
	else
		antflags="${antflags} -Dmaven.test.skip=true"
	fi

	local cp

	for atom in ${gcp}; do
		cp+=":$(java-pkg_getjars ${getjarsarg} ${atom})"
	done
```
At the end, `java-pkg_getjars` cannot find `ant-junit` as a package because it is now provided by `dev-java/ant[junit]` as `/usr/share/ant/lib/ant-junit.jar`. 
An approach to solve this error might be to stop adding `ant-junit` to `gcp`.

**Pull request closed without merging**